cfme/tests/intelligence/reports/test_reports.py

```diff
@@ -0,0 +1,45 @@
+import pytest
+
+from cfme import test_requirements
+from cfme.rest.gen_data import users as _users
+from cfme.utils.rest import assert_response
+
+pytestmark = [test_requirements.report, pytest.mark.tier(3), pytest.mark.sauce]
+
+
+@pytest.fixture(scope="function")
+def rbac_api(appliance, request):
+    user, user_data = _users(
+        request, appliance, password="smartvm", group="EvmGroup-user"
+    )
+    api = appliance.new_rest_api_instance(
+        entry_point=appliance.rest_api._entry_point,
+        auth=(user[0].userid, user_data[0]["password"]),
+    )
+    assert_response(api)
+    yield api
+    appliance.rest_api.collections.users.action.delete(id=user[0].id)
+
+
+@test_requirements.rest
+@pytest.mark.tier(1)
+def test_non_admin_user_reports_access_rest(appliance, request, rbac_api):
+    """ This test checks if a non-admin user with proper privileges can access all reports via API.
+
+    Polarion:
+        assignee: pvala
+        casecomponent: Rest
+        caseimportance: medium
+        initialEstimate: 1/12h
+        tags: report
+        setup:
+            1. Create a user with privilege to access reports and REST.
+            2. Instantiate a MiqApi instance with the user.
+        testSteps:
+            1. Access all reports with the new user with the help of newly instantiated API.
+        expectedResults:
+            1. User should be able to access all reports.
+    """
+    report_data = rbac_api.collections.reports.all
+    assert_response(appliance)
```

Comment:
QUESTION: @valaparthvi, what is that for ?

Reply:
That is to check if the last request sent via REST API received a successful 200 OK response.
Last request sent here is: `rbac_api.collections.reports.all`